node/service/src/service/mod.rs

```diff
@@ -115,3 +76,474 @@ fn set_prometheus_registry(config: &mut sc_service::Configuration) -> ServiceRes
+		Some(shared_authority_set.clone()),
+	);
+	let rpc_extensions_builder = Box::new({
+		let client = client.clone();
+		let keystore = keystore_container.sync_keystore();
+		let transaction_pool = transaction_pool.clone();
+		let select_chain = select_chain.clone();
+		let chain_spec = config.chain_spec.cloned_box();
+		let shared_voter_state = shared_voter_state.clone();
+		let network = network.clone();
+
+		move |deny_unsafe, subscription_executor: SubscriptionTaskExecutor| -> RpcServiceResult {
+			let deps = FullDeps {
+				client: client.clone(),
+				pool: transaction_pool.clone(),
+				select_chain: select_chain.clone(),
+				chain_spec: chain_spec.cloned_box(),
+				deny_unsafe,
+				babe: BabeDeps {
+					babe_config: babe_config.clone(),
+					shared_epoch_changes: shared_epoch_changes.clone(),
+					keystore: keystore.clone(),
+				},
+				grandpa: GrandpaDeps {
+					shared_voter_state: shared_voter_state.clone(),
+					shared_authority_set: shared_authority_set.clone(),
+					justification_stream: justification_stream.clone(),
+					subscription_executor: subscription_executor.clone(),
+					finality_proof_provider: finality_proof_provider.clone(),
+				},
+				eth: EthDeps {
+					config: eth_rpc_config.clone(),
+					graph: transaction_pool.pool().clone(),
+					is_authority,
+					network: network.clone(),
+					filter_pool: filter_pool.clone(),
+					backend: dvm_backend.clone(),
+					fee_history_cache: fee_history_cache.clone(),
+					overrides: overrides.clone(),
+					block_data_cache: block_data_cache.clone(),
+					rpc_requesters: eth_rpc_requesters.clone(),
+				},
+			};
+
+			darwinia_rpc::create_full(deps, subscription_task_executor.clone()).map_err(Into::into)
+		}
+	});
+	let rpc_handlers = sc_service::spawn_tasks(SpawnTasksParams {
+		config,
+		backend: backend.clone(),
+		client: client.clone(),
+		keystore: keystore_container.sync_keystore(),
+		network: network.clone(),
+		rpc_extensions_builder,
+		transaction_pool: transaction_pool.clone(),
+		task_manager: &mut task_manager,
+		on_demand: None,
+		remote_blockchain: None,
+		system_rpc_tx,
+		telemetry: telemetry.as_mut(),
+	})?;
+
+	if is_authority {
+		let can_author_with = CanAuthorWithNativeVersion::new(client.executor().clone());
+		let proposer = ProposerFactory::new(
+			task_manager.spawn_handle(),
+			client.clone(),
+			transaction_pool,
+			prometheus_registry.as_ref(),
+			telemetry.as_ref().map(|x| x.handle()),
+		);
+		let slot_duration = babe_link.config().slot_duration();
+		let babe_config = BabeParams {
+			keystore: keystore_container.sync_keystore(),
+			client: client.clone(),
+			select_chain,
+			block_import: babe_import,
+			env: proposer,
+			sync_oracle: network.clone(),
+			justification_sync_link: network.clone(),
+			create_inherent_data_providers: move |_, ()| async move {
+				// TODO: https://github.com/paritytech/polkadot/pull/5750
+				let uncles = sc_consensus_uncles::create_uncles_inherent_data_provider(
```

Comment:
After discussing with @andresilva, I think this could be removed.

See also

- [ ] https://github.com/paritytech/polkadot/pull/5750

We can wait for the Polkadot PR.

Reply:
Crab
- https://crab.subscan.io/extrinsic?address=&module=authorship&call=set_uncles&result=all&signedChecked=all
https://github.com/darwinia-network/darwinia/blob/main/runtime/crab/src/lib.rs#L170

Pangolin
- https://pangolin.subscan.io/extrinsic?address=&module=authorship&call=set_uncles&result=all&signedChecked=all
- https://github.com/darwinia-network/darwinia-common/blob/main/node/runtime/pangolin/src/lib.rs#L162

As you can see, we don't construct the `Inherent` for Darwinia-like runtimes. And there are no `set_uncles` on the live chain.